elyra/pipeline/processor.py

```diff
@@ -69,22 +70,28 @@ def __init__(self, **kwargs):
                 # log and ignore initialization errors
                 self.log.error('Error registering processor "{}" - {}'.format(processor, err))
 
-    async def process(self, pipeline):
-        processor_type = pipeline.runtime
+    def _get_processor_for_runtime(self, processor_type: str):
         processor = self._registry.get_processor(processor_type)
 
         if not processor:
-            raise RuntimeError('Could not find pipeline processor for [{}]'.format(pipeline.runtime))
+            raise RuntimeError('Could not find pipeline processor for [{}]'.format(processor_type))
```

Comment:
Should `get_processor()` raise instead?  It looks like this kind of condition is used elsewhere.  

If we need to _validate_ that a given type has an associated processor, we should look into implementing `__contains__` (similar to what [`MultiKernelManager` does to check if a given `KernelManager` is present](https://github.com/jupyter/jupyter_client/blob/98faaf91d00b4fbf17b2fbf0b08b1fd23adc27bf/jupyter_client/multikernelmanager.py#L164-L165)).

Reply:
I'll move the raise for now and put `__contains__` into my "to-discuss" list!